core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -635,6 +635,96 @@ public UserCancelledException(String message, Throwable cause, String recoverySu
         }
     }
 
+    /**
+     * Could not perform the operation due to incorrect lambda configuration.
+     */
+    public static class UnexpectedLambdaException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Invalid lambda response error.";
+        private static final String RECOVERY_SUGGESTION =
+                "Make sure that the lambda configuration is correct.";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public UnexpectedLambdaException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     * Could not perform the operation due to incorrect lambda configuration.
+     */
+    public static class UserLambdaValidationException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Invalid lambda response error.";
+        private static final String RECOVERY_SUGGESTION =
+                "Make sure that the lambda configuration is correct.";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public UserLambdaValidationException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     * Could not perform the operation due to invalid sms role trust relationship.
+     */
+    public static class InvalidSmsRoleTrustRelationshipException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Invalid SMS Role Trust Relationship error.";
+        private static final String RECOVERY_SUGGESTION =
```

Comment:
Where is this used?

Reply:
It is not used anywhere but these are list of exceptions that could occur when calling the updateuserattribute api. The cognito triggers the error, and this is just for our reference.

Reply:
Is this necessary then? wouldn't the default exception wrap the cognitos error if it cannot find the mapping?

Reply:
Yes I agree but the iOS team also has similar implementation. It is possible that going forward that we want to customize the message.

Reply:
Is the error message `Invalid SMS Role Trust Relationship error.` returned by Cognito directly or defined by us knowing how the API works?

Reply:
It will be returned by cognito directly, so perhaps we don't need to declare it. I think the AuthException file contains all exceptions encountered during auth calls and we can customize as we please.